site/src/sphinx/client-custom-http-headers.rst

```diff
@@ -78,8 +78,8 @@ If you want more freedom on how you manipulate the request headers, use a decora
 
     // Add a decorator that inserts the custom header.
     cb.decorator((delegate, ctx, req) -> { // See DecoratingHttpClientFunction and DecoratingRpcClientFunction.
-        req.headers().set(AUTHORIZATION, credential);
-        return delegate.execute(ctx, req);
+        RequestHeaders req0 = req.headers().set(AUTHORIZATION, credential).toBuilder().build();
+        return delegate.execute(ctx, HttpRequest.of(req0, req));
```

Comment:
Ooh - this should be `HttpRequest.of(req, authorized)`. But it's confusing indeed that we have many methods that take `newHeaders` as the first parameter but this one takes it second and they both compile. I have filed #2278